app/javascript/react/components/Map/Map.tsx

```diff
@@ -93,59 +122,126 @@ const Map = () => {
       const east = bounds.getNorthEast().lng();
       const west = bounds.getSouthWest().lng();
       setMapBounds({ north, south, east, west });
-      setCurrentZoom(map.getZoom() || DEFAULT_ZOOM);
     },
     [mapInstance]
   );
 
-  const handleMarkerClick = (streamId: React.SetStateAction<number | null>) => {
-    if (mapInstance) {
-      setPreviousZoom(mapInstance.getZoom() || DEFAULT_ZOOM);
-      setPreviousCenter(
-        mapInstance.getCenter()?.toJSON() || DEFAULT_MAP_CENTER
-      );
+  // Handlers
+  const handleMarkerClick = (streamId: number | null, id: number | null) => {
+    if (streamId) {
+      fixedSessionTypeSelected
+        ? dispatch(fetchFixedStreamById(streamId))
+        : dispatch(fetchMobileStreamById(streamId));
+    }
+
+    if (!selectedStreamId) {
+      setSelectedSessionId(id);
+      setSelectedStreamId(streamId);
+      setModalOpen(false);
+      setTimeout(() => {
+        setModalOpen(true);
+      }, 0);
```

Comment:
if the timeout is 0 can we just use `setModalOpen(true)`?

Reply:
This is still valid solution even with 0  -> https://medium.com/@madhavkhera868/article-on-zero-delay-settimeout-function-in-javascript-f25a08ca350b